src/backend.rs

```diff
@@ -318,6 +318,31 @@ pub trait Backend {
 
     /// Flush any buffered content to the terminal screen.
     fn flush(&mut self) -> io::Result<()>;
+
+    /// Scroll a region of the screen upwards, where a region is a range of rows.
+    ///
+    /// Each row in the region is replaced by the row `amount` rows below it, except the bottom
+    /// `amount` rows, which are replaced by empty rows. If `amount` is equal to or larger than the
+    /// number of rows in the region, then all rows are replaced with empty rows.
+    ///
+    /// If the region includes row 0, then `amount` rows are copied into the bottom of the
+    /// scrollback buffer. These rows are first taken from the old contents of the region, starting
+    /// from the top. If there aren't sufficient rows in the region, then the remainder are empty
+    /// rows.
+    ///
+    /// The position of the cursor afterwards is undefined.
+    #[cfg(feature = "scrolling-regions")]
+    fn scroll_region_up(&mut self, region: std::ops::Range<u16>, amount: u16) -> io::Result<()>;
```

Comment:
Sure.

Reply:
I referred the reader to look at the tests for the test backend, as I think it would make the doc comment very large to have enough examples to illustrate all of the edge cases.